### PR TITLE
Add deprecation warnings to geo fields that define multifields

### PR DESCRIPTION
**Pull request by romseygeek**
With the exception of `geo_point`, no geometry field mappers actually support
the use of multifields.  However, we happily accept multifield definitions on these
mappers and silently ignore them when it comes to indexing documents.  With
this commit, we emit deprecation warnings when a multifield definition is included 
on a geometry field mapper that does not in fact support one.

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Review by jtibshirani — APPROVED** [review]
It's great to clean this up!

**Review by iverase — APPROVED** [review]

**Comment by romseygeek** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java [hunk 1]
```diff
@@ -83,6 +87,13 @@ public Builder ignoreZValue(boolean ignoreZValue) {
 
         @Override
         public GeoShapeFieldMapper build(ContentPath contentPath) {
+            if (multiFieldsBuilder.hasMultiFields()) {
+                DEPRECATION_LOGGER.deprecate(
+                    DeprecationCategory.MAPPINGS,
+                    "geo_shape_multifields",
+                    "Adding multifields to geo_shape mappers has no effect and will be forbidden in future"
```
[the remark this answers is not in the document]
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java]: ++ have updated accordingly

**Pull request merged**